### ISSUE TITLE
Upgrading from 2.12.2 to 2.13.1; awx-postgres-15-0 stuck in pending 

### ISSUE THREAD

**Issue by cpilton** [ansible/tower-operator#1772]
### Please confirm the following

- [X] I agree to follow this project's [code of conduct](https://docs.ansible.com/ansible/latest/community/code_of_conduct.html).
- [X] I have checked the [current issues](https://github.com/ansible/awx-operator/issues) for duplicates.
- [X] I understand that the AWX Operator is open source software provided for free and that I might not receive a timely response.

### Bug Summary

awx-postgres-15-0 is stuck in pending after upgrading to 2.13.1.
```
Warning  FailedScheduling  14m (x2 over 19m)  default-scheduler  0/1 nodes are available: pod has unbound immediate PersistentVolumeClaims. preemption: 0/1 nodes are available: 1 Preemption is not helpful for scheduling..
```

PVs:
```
NAME                                       CAPACITY   ACCESS MODES   RECLAIM POLICY   STATUS   CLAIM                               STORAGECLASS          REASON   AGE
awx-projects-volume                        2Gi        RWO            Retain           Bound    awx/awx-projects-claim              awx-projects-volume            547d
awx-postgres-13-volume                     6Gi        RWO            Retain           Bound    awx/postgres-13-awx-postgres-13-0   awx-postgres-volume            547d
auto-pv                                    2Gi        RWO            Retain           Bound    awx/auto-pvc                        auto-storage                   274d
pvc-8fb18ecb-5b74-4f8e-af88-172c1cddab15   5Gi        RWO            Delete           Bound    awx/awx-backup-claim                local-path                     59m
```

PVCs:
```
NAME                            STATUS    VOLUME                                     CAPACITY   ACCESS MODES   STORAGECLASS          AGE
awx-projects-claim              Bound     awx-projects-volume                        2Gi        RWO            awx-projects-volume   547d
postgres-13-awx-postgres-13-0   Bound     awx-postgres-13-volume                     6Gi        RWO            awx-postgres-volume   547d
auto-pvc                        Bound     auto-pv                                    2Gi        RWO            auto-storage          274d
awx-backup-claim                Bound     pvc-8fb18ecb-5b74-4f8e-af88-172c1cddab15   5Gi        RWO            local-path            59m
postgres-15-awx-postgres-15-0   Pending                                                                        awx-postgres-volume   26m
```

The awx-operator-controller-manager loops around here:
```
--------------------------- Ansible Task StdOut -------------------------------

TASK [installer : Create Database if no database is specified] *****************
task path: /opt/ansible/roles/installer/tasks/upgrade_postgres.yml:33

-------------------------------------------------------------------------------
{"level":"info","ts":"2024-03-14T11:31:52Z","logger":"proxy","msg":"Read object from cache","resource":{"IsResourceRequest":true,"Path":"/apis/apps/v1/namespaces/awx/statefulsets/awx-postgres-15","Verb":"get","APIPrefix":"apis","APIGroup":"apps","APIVersion":"v1","Namespace":"awx","Resource":"statefulsets","Subresource":"","Name":"awx-postgres-15","Parts":["statefulsets","awx-postgres-15"]}}
{"level":"info","ts":"2024-03-14T11:31:54Z","logger":"proxy","msg":"Read object from cache","resource":{"IsResourceRequest":true,"Path":"/api/v1/namespaces/awx/pods","Verb":"list","APIPrefix":"api","APIGroup":"","APIVersion":"v1","Namespace":"awx","Resource":"pods","Subresource":"","Name":"","Parts":["pods"]}}
```

Eventually - this times out:
```
 TASK [Create Database if no database is specified] ******************************** 
fatal: [localhost]: FAILED! => {"changed": false, "duration": 120, "method": "apply", "msg": "\"StatefulSet\" \"awx-postgres-15\": Timed out waiting on resource"
```

The task and web pods never get created.

### AWX Operator version

2.13.1

### AWX version

24.0.0

### Kubernetes platform

kubernetes

### Kubernetes/Platform version

k3s version v1.28.7+k3s1 (051b14b2)

### Modifications

no

### Steps to reproduce

`./kustomize` build . | kubectl apply -f -`

with:

```
apiVersion: kustomize.config.k8s.io/v1beta1
kind: Kustomization
resources:
  # Find the latest tag here: https://github.com/ansible/awx-operator/releases
  - github.com/ansible/awx-operator/config/default?ref=2.13.1

# Set the image tags to match the git version from above
images:
  - name: quay.io/ansible/awx-operator
    newTag: 2.13.1

# Specify a custom namespace in which to install AWX
namespace: awx
```

### Expected results

The new PVC is created and and binded to the new postgres pod

### Actual results

The new PVC is created and but not binded to the new postgres pod. 
The installation bails out.

**Comment by cpilton** [comment]
PVC config is as follows:  

```
---
apiVersion: v1
kind: PersistentVolumeClaim
metadata:
  name: awx-projects-claim
spec:
  accessModes:
    - ReadWriteOnce
  volumeMode: Filesystem
  resources:
    requests:
      storage: 2Gi
  storageClassName: awx-projects-volume
```

PV config is as follows:

```
---
apiVersion: v1
kind: PersistentVolume
metadata:
  name: awx-postgres-13-volume
spec:
  accessModes:
    - ReadWriteOnce
  persistentVolumeReclaimPolicy: Retain
  capacity:
    storage: 6Gi
  storageClassName: awx-postgres-volume
  hostPath:
    path: /mqawx/postgres-13

---
apiVersion: v1
kind: PersistentVolume
metadata:
  name: awx-projects-volume
spec:
  accessModes:
    - ReadWriteOnce
  persistentVolumeReclaimPolicy: Retain
  capacity:
    storage: 2Gi
  storageClassName: awx-projects-volume
  hostPath:
    path: /mqawx/projects
```

**Comment by kunkanon** [comment]
I have the same issue.

Kubernets v1.25.6+k3s1
Kustomize Version: v4.5.7

[operator.log.zip](https://github.com/ansible/awx-operator/files/14601416/operator.log.zip)

![Captura de Tela 2024-03-14 às 08 36 11](https://github.com/ansible/awx-operator/assets/23107622/42b4cd1b-dbcb-4342-955f-4bfdda756899)

**Comment by cpilton** [comment]
I have created a new PV in the PV config as follows: 

```
---
apiVersion: v1
kind: PersistentVolume
metadata:
  name: awx-postgres-15-volume
spec:
  accessModes:
    - ReadWriteOnce
  persistentVolumeReclaimPolicy: Retain
  capacity:
    storage: 6Gi
  storageClassName: awx-postgres-volume
  hostPath:
    path: /mqawx/postgres-15

---
```

The PVC is automatically binding to this now...

However, the pod is crashing upon start:
```
  Normal   SandboxChanged    23s                    kubelet            Pod sandbox changed, it will be killed and re-created.
  Normal   Pulled            3s (x2 over 21s)       kubelet            Container image "quay.io/sclorg/postgresql-15-c9s:latest" already present on machine
  Normal   Created           3s (x2 over 21s)       kubelet            Created container postgres
  Normal   Started           3s (x2 over 20s)       kubelet            Started container postgres
  Warning  BackOff           2s (x3 over 18s)       kubelet            Back-off restarting failed container postgres in pod awx-postgres-15-0_awx(70c2d5aa-6689-4985-a5c1-d42cedee8eb5)
```

The pod logs are no longer empty: 
`mkdir: cannot create directory '/var/lib/pgsql/data/userdata': Permission denied`

**Comment by cpilton** [comment]
Based on the above, i suggest this is now a Duplicate of #1770

**Comment by kurokobo** [comment]
@cpilton 

```bash
sudo mkdir -p /mqawx/postgres-15/data
sudo chmod 700 /mqawx/postgres-15/data
sudo chown 26:0 /mqawx/postgres-15/data
```

**Comment by kurokobo** [comment]
Your deployment seems to be basically based on [my guide](https://github.com/kurokobo/awx-on-k3s). Now I'm writing this very part to update upgrading guide. This is the snippet in my draft.

Note, this is applicable for the deployment that based on [my guide](https://github.com/kurokobo/awx-on-k3s) only.

> [!WARNING]
> If you are planning to upgrade AWX Operator **from `2.12.2` or earlier to `2.13.1` or later**, note that since the bundled PostgreSQL version will be changed to 15, so the following additional tasks are required.
>
> ```bash
> # Required only when upgrading from 2.12.2 or earlier to 2.13.1 or later
> sudo mkdir -p /data/postgres-15/data
> sudo chmod 700 /data/postgres-15/data
> sudo chown 26:0 /data/postgres-15/data
> cat <<EOF > pv-postgres-15.yaml
> ---
> apiVersion: v1
> kind: PersistentVolume
> metadata:
>   name: awx-postgres-15-volume
> spec:
>   accessModes:
>     - ReadWriteOnce
>   persistentVolumeReclaimPolicy: Retain
>   capacity:
>     storage: 8Gi
>   storageClassName: awx-postgres-volume
>   hostPath:
>     path: /data/postgres-15
> EOF
> kubectl apply -f pv-postgres-15.yaml
> ```

**Comment by cpilton** [comment]
@kurokobo Thank you. Your suggestion worked and AWX is running again

**Comment by cuspofaries** [comment]
> I have the same issue.
> 
> Kubernets v1.25.6+k3s1 Kustomize Version: v4.5.7
> 
> [operator.log.zip](https://github.com/ansible/awx-operator/files/14601416/operator.log.zip)
> 
> ![Captura de Tela 2024-03-14 às 08 36 11](https://private-user-images.githubusercontent.com/23107622/312821607-42b4cd1b-dbcb-4342-955f-4bfdda756899.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTA3ODczMjAsIm5iZiI6MTcxMDc4NzAyMCwicGF0aCI6Ii8yMzEwNzYyMi8zMTI4MjE2MDctNDJiNGNkMWItZGJjYi00MzQyLTk1NWYtNGJmZGRhNzU2ODk5LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMTglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzE4VDE4MzcwMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWUyNWQzMmUyYWNlZWRjOWY3NjNkZDViMGZiNjhjZDI5N2ViZDk1NWVkYTc5ZDI3YTZlZmExYTQ4MDk1OTJkNTgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.7csYrrktgokZBx_QH7DWvTNUSDzNJspt7sLwjnWd-yc)

Did you manage to fix this ? I have the exacte same issue

**Comment by kunkanon** [comment]
> ge to fix this ? I have the exacte same i

No. As I can't keep the server down, I made all the backups and performed a clean installation of the latest version available.

**Comment by AnthonyMacle** [comment]
> > ge to fix this ? I have the exacte same i
> 
> No. As I can't keep the server down, I made all the backups and performed a clean installation of the latest version available.

Well, I guess I am going to do this as well. Did you make a dump of the Postgres-13 and then you imported it into Postgres-15 ?

**Comment by DanyOtty** [comment]
@kurokobo: Thank you to saving my day!
